symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php

```diff
@@ -23,43 +23,40 @@
  * Description of GoogleAuthProvider
  */
 class GoogleAuthProvider extends AbstractAuthProvider {
-
     const PROFILE_SCOPE = 'profile';
     const EMAIL_SCOPE = 'email';
+    protected $loginService;
 
-    public function validateUser($provider, $authProvider = null) {
-
+    /**
+     * @param OpenidProvider $provider
+     * @param AuthProviderExtraDetails $authProvider
+     * @return array
+     */
+    public function validateUser($provider, $authProvider = null)
+    {
         $gClient = new Google_Client();
         $gClient->setApplicationName('Login to Google +');
         $gClient->setClientId($authProvider->getClientId());
         $gClient->setClientSecret($authProvider->getClientSecret());
         $gClient->setRedirectUri($provider->getProviderUrl());
         $gClient->setDeveloperKey($authProvider->getDeveloperKey());
-        $gClient->addScope(array(self::EMAIL_SCOPE,self::PROFILE_SCOPE));
-
-        $gService = new Google_Service_Plus($gClient);
+        $gClient->addScope(array(self::EMAIL_SCOPE, self::PROFILE_SCOPE));
 
         if (isset($_GET['code'])) {
-            $gClient->authenticate($_GET['code']);
+            $gClient->fetchAccessTokenWithAuthCode($_GET['code']);
```

Comment:
It's not good to couple GET params inside the method.
better to pass it as parameter or setter method for the class